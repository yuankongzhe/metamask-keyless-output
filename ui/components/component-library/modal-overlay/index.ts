export { ModalOverlay } from './modal-overlay';
export type {
  ModalOverlayProps,
  ModalOverlayComponent,
  ModalOverlayStyleUtilityProps,
} from './modal-overlay.types';
