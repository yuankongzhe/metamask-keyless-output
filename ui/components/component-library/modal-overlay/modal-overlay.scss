@use "design-system";

.mm-modal-overlay {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: design-system.$modal-z-index;
}
