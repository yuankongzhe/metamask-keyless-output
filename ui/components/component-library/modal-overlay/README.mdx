import { Story, Canvas, ArgsTable } from '@storybook/addon-docs';

import { ModalOverlay } from './modal-overlay';

# ModalOverlay

`ModalOverlay` is a transparent overlay that covers the entire screen. It is used to dim the background when a modal is open.

[MetaMask Design System Guides](https://www.notion.so/MetaMask-Design-System-Guides-Design-f86ecc914d6b4eb6873a122b83c12940)

<Canvas>
  <Story id="components-componentlibrary-modaloverlay--default-story" />
</Canvas>

## Props

<ArgsTable of={ModalOverlay} />

### On Click

Use the `onClick` prop to handle clicks on the overlay

<Canvas>
  <Story id="components-componentlibrary-modaloverlay--on-click" />
</Canvas>

```jsx
import React, { useState } from 'react';
import { ModalOverlay } from '../../component-library';

const [open, setOpen] = useState(false);
const handleOnClick = () => {
  setOpen(!open);
};

<button onClick={handleOnClick}>Show modal overlay</button>;
{
  open && <ModalOverlay onClick={handleOnClick} />;
}
```
