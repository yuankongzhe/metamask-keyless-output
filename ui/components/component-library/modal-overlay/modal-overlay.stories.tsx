import React, { useState } from 'react';
import { StoryFn, Meta } from '@storybook/react';

import { ModalOverlay } from './modal-overlay';

import README from './README.mdx';

export default {
  title: 'Components/ComponentLibrary/ModalOverlay',
  component: ModalOverlay,
  parameters: {
    docs: {
      page: README,
    },
  },
  argTypes: {
    className: {
      control: 'text',
    },
    onClick: {
      action: 'onClick',
    },
  },
} as Meta<typeof ModalOverlay>;

const Template: StoryFn<typeof ModalOverlay> = (args) => (
  <ModalOverlay {...args} />
);

export const DefaultStory = Template.bind({});
DefaultStory.storyName = 'Default';

export const OnClick: StoryFn<typeof ModalOverlay> = (args) => {
  const [open, setOpen] = useState(false);
  const handleOnClick = () => {
    setOpen(!open);
  };
  return (
    <>
      <button onClick={handleOnClick}>Show modal overlay</button>
      {open && <ModalOverlay {...args} onClick={handleOnClick} />}
    </>
  );
};
