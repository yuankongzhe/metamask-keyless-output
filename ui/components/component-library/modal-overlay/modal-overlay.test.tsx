import React from 'react';
import { render } from '@testing-library/react';

import { ModalOverlay } from './modal-overlay';

describe('ModalOverlay', () => {
  it('should render ModalOverlay without error', () => {
    const { getByTestId } = render(
      <ModalOverlay data-testid="modal-overlay" />,
    );
    expect(getByTestId('modal-overlay')).toBeDefined();
    expect(getByTestId('modal-overlay')).toHaveClass('mm-modal-overlay');
  });
  it('should match snapshot', () => {
    const { container } = render(<ModalOverlay />);
    expect(container).toMatchSnapshot();
  });
  it('should render with and additional className', () => {
    const { getByTestId } = render(
      <ModalOverlay data-testid="modal-overlay" className="test-class" />,
    );
    expect(getByTestId('modal-overlay')).toHaveClass('test-class');
  });
  it('should fire the onClick function when clicked', () => {
    const onClick = jest.fn();
    const { getByTestId } = render(
      <ModalOverlay data-testid="modal-overlay" onClick={onClick} />,
    );
    getByTestId('modal-overlay').click();
    expect(onClick).toHaveBeenCalled();
  });
});
