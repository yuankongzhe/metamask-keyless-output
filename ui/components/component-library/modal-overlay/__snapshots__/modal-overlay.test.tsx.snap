// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalOverlay should match snapshot 1`] = `
<div>
  <div
    aria-hidden="true"
    class="mm-box mm-modal-overlay mm-box--width-full mm-box--height-full mm-box--background-color-overlay-default"
  />
</div>
`;
